src/lib/peer.ts

```diff
@@ -4,24 +4,30 @@ import { BaseAPI } from './api'
 import { ProfileEntity } from './types'
 
 export class PeerAPI extends BaseAPI {
-  cache: Record<string, Profile> = {}
+  cache: Record<string, Promise<Profile>> = {}
   lambdasClient: LambdasClient
 
   constructor(url: string) {
     super(url)
     this.lambdasClient = new LambdasClient(`${url}/lambdas`)
   }
 
-  // TODO, how many ways of retrieving a profile are out there?
+  /**
+   * Fetches a profile only once by caching the promise
+   * to prevent concurrent flying requests
+   *
+   * @param address - The address of the profile to retrieve.
+   */
   public async fetchProfile(address: string): Promise<Profile> {
     if (address in this.cache) {
       return this.cache[address]
     }
-    const profiles = await this.lambdasClient.fetchProfiles([
-      address.toLowerCase()
-    ])
-    this.cache[address] = profiles[0]
-    return profiles[0]
+
+    this.cache[address] = this.lambdasClient
+      .fetchProfiles([address.toLowerCase()])
+      .then()
```

Comment:
Is this last `.then()` necessary?

Reply:
Oh yes! It is! Sorry, I forgot to write the whole line. We need it so we can retrieve only one profile